packages/block-library/src/image/edit.native.js

```diff
@@ -461,6 +468,45 @@ export class ImageEdit extends Component {
 		} );
 	}
 
+	onSelectURL( newURL ) {
+		const {
+			createErrorNotice,
+			imageDefaultSize,
+			setAttributes,
+		} = this.props;
+
+		if ( isURL( newURL ) ) {
+			this.setState( {
+				isFetchingImage: true,
+			} );
+
+			// Use RN's Image.getSize to determine if URL is a valid image
+			RNImage.getSize(
+				newURL,
+				( width, height ) => {
+					setAttributes( {
+						url: newURL,
+						id: undefined,
+						width,
+						height,
```

Comment:
I saw that in the web version `width` and `height` attributes are set as `undefined.

https://github.com/WordPress/gutenberg/blob/97474fa8060a4d37081f431c5cf4e54306e66746/packages/block-library/src/image/edit.js#L285-L286

Not sure if it's totally related to this, but I noticed when setting a big image in both versions, that adding an URL of a big image in the native version is not properly rendered in the web version. Probably we should match the same logic between web and native on this part, in order to prevent potential side effects.

<img width="1746" alt="Screenshot 2022-05-19 at 19 13 02" src="https://user-images.githubusercontent.com/14905380/169359396-055aaae8-7ec3-4515-83c7-cfb72c673592.png">

Reply:
Updated - I agree this change definitely belongs in this PR after testing on web and native. The dimension values should be `undefined` to match the web behavior, and they were indeed that when I initially created this PR. This was an oversight on my part when I added the `getSize` function, which returned width and height values by default. ([#](https://github.com/WordPress/gutenberg/pull/40334/commits/7995ef9976319757cb865100ea6779fe1c9c7f9b#diff-7126ae03000a022bcefa0d2a5fb03b6aca56ed4047e3a49d2131babf229cf6c1L479-L480))